src/main/java/org/example/bookshop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,134 @@
+import org.example.bookshop.exception.EntityNotFoundException;
+import org.example.bookshop.mapper.OrderItemMapper;
+import org.example.bookshop.mapper.OrderMapper;
+import org.example.bookshop.model.Order;
+import org.example.bookshop.model.OrderItem;
+import org.example.bookshop.model.ShoppingCart;
+import org.example.bookshop.model.User;
+import org.example.bookshop.repository.CartItemRepository;
+import org.example.bookshop.repository.CartRepository;
+import org.example.bookshop.repository.OrderItemRepository;
+import org.example.bookshop.repository.OrderRepository;
+import org.example.bookshop.service.OrderService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final CartItemRepository cartItemRepository;
+    private final CartRepository cartRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderDto add(OrderPurchaseRequestDto requestDto, User user) {
+        ShoppingCart shoppingCart = cartRepository.findByUserId(
+                user.getId()).orElseThrow(() -> new EntityNotFoundException(
+                "Can't find order for user: " + user.getEmail()));
+        Order order = orderMapper.toEntity(shoppingCart);
+        order.setShippingAddress(requestDto.getShippingAddress());
+        order.setOrderDate(LocalDateTime.now());
+        order.setStatus(Order.Status.PENDING);
+        Set<OrderItem> orderItems = findOrderItems(user);
+        orderItems.forEach(item ->
+                item.setPrice(item.getPrice().multiply(new BigDecimal(item.getQuantity()))));
+        order.setTotal(getTotalPrice(orderItems));
+        orderRepository.save(order);
+        orderItems.forEach(item -> {
+            item.setOrder(order);
+            orderItemRepository.save(item);
+        });
+        order.setOrderItems(orderItems);
+        shoppingCart.getCartItems().clear();
+        cartRepository.save(shoppingCart);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public Set<OrderDto> getAll(User user, Sort sort, Pageable pageable) {
+        return orderRepository.findAllByUserId(user.getId(), pageable).stream()
+                .map(orderMapper::toDto)
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public void update(OrderStatusRequestDto requestDto, Long orderId, User user) {
+        Order order = findOrderByIdAndUserId(orderId, user.getId());
+        Order.Status status = Order.Status.valueOf(requestDto.getStatus());
+        if (order.getStatus().equals(status)) {
+            throw new DataProcessingException("Order already has status: " + status);
+        }
```

Comment:
I don’t see your global exception handler but I guess it provide internal server error. is it ok?

Reply:
There is provides 401 http response

Reply:
Okay, your filter chain does it. Anyway, you see an exception in the console. Is it ok? 
401 Unauthorized  - is it the appropriate status for this exception? 
You should handle this exception and return CONFLICT.